### PR TITLE
Server-side physics calculations

### PR DESCRIPTION
**Pull request by filiptrplan**
* implement basic physics loop that runs `gameFPS` times per second(runs at 60.00FPS due to microsecond precision 😎😎)
* add basic velocity calculations server-side
* add interpolation between the server position and the client position

### CODE REVIEW

File: world-client/src/state/player.js [hunk 1]
```diff
@@ -10,6 +12,11 @@ export default class Player {
     this.speed = this.defaultSpeed;
     this.turnSpeed = 5;
 
+    this.serverPosition = { x: x, y: y};
+    this.onMessagePosition = {x: x, y: y}
+    this.messageReceiveTime = 0;
+    this.desiredInterpTime = 1000 / 60 - 1;
```
Comment (aJuvan) [inline world-client/src/state/player.js:22]: Create an ENV variable for gameticks, so we can change it depending on our needs. Also take a looks at the [dotenv package](https://www.npmjs.com/package/dotenv) to add default values to a file.

Comment (aJuvan) [reply world-client/src/state/player.js:22]: Also relates to the backend

File: world-server/src/world/World.ts [hunk 2]
```diff
@@ -58,6 +64,64 @@ export class World {
...
+    setupGameTick(){
+        let average = 0;
+        // Send periodic ticks per second
+        let debugTicks = false;
+        let notificationInterval = 20;
+        if(debugTicks){
+            setInterval(() => {
+                console.log(
+                    "Ticks per second: " + (this.ticks / notificationInterval).toFixed(2)
+                );
```
Comment (aJuvan) [inline world-server/src/world/World.ts]: Remove debug log

File: world-server/src/world/Math.ts [hunk 3]
```diff
@@ -2,3 +2,11 @@ export interface Vector2 {
     x: number;
     y: number;
 }
+
+export function vectorAdd(vec1: Vector2, vec2: Vector2): Vector2 {
```
Comment (jakakonda) [inline world-server/src/world/Math.ts:6]: Would suggest using as static functions inside Vector2 if possible (less imports, quasi namespaced). This is if it's possible to convert to abstract class without breaking changes.

**Review by aJuvan — CHANGES_REQUESTED** [review]

File: world-server/src/world/Math.ts (hunk 3 above)
Comment (filiptrplan) [reply world-server/src/world/Math.ts:6]: The Vector2 interface is used in message handling and it is an interface precisely because the client sends simple objects with no type.
tl;dr typescript complains

File: world-server/src/world/World.ts (hunk 2 above)
Comment (filiptrplan) [reply world-server/src/world/World.ts]: @davidmohar said to keep it, we may use the code later down the line when monitoring.

Comment (davidmohar) [reply world-server/src/world/World.ts]: @1234filip I would implement monitoring differently :) So we can remove this `console.log()`

**Review by aJuvan — APPROVED** [review]
